### ISSUE TITLE
We may want/need to move away from Keybase

### ISSUE THREAD

**Issue by zdb999** [sandstorm-io/sandstorm#3329]
Keybase has just announced that they have been [acquired by Zoom](https://keybase.io/blog/keybase-joins-zoom), a company which does not have a good privacy record. They have made no commitments about continuing to maintain Keybase, and their blog post indicates that this is an aquihire that will deprioritize the Keybase platform. In the mid-term we should consider moving away from Keybase, or at least offering alternatives for package signing.

**Comment by ocdtrekkie** [comment]
Yeah, we discussed this a bit on IRC yesterday: https://freenode.irclog.whitequark.org/sandstorm/2020-05-07

Fundamentally at least for now, Keybase's existing security is probably not significantly compromised by having been acquired, the attestations of identity people have posted are still verifiable, and I believe if a user's identities change at all, people following them are notified.

But I suspect Keybase will not exist long, so a replacement is probably worthwhile. Alternatives like keys.pub should probably be researched. I think the ideal goal is that we have something that can check identity attestations like Keybase does hosted within a Sandstorm grain. Such a grain could be hosted on the same server as the App Index for Sandstorm servers to query, such that everyone's individual Sandstorm servers don't have to reach out and talk to GitHub/Twitter/etc. but ideally that people can still manually check to see those attestations remain valid.

**Comment by ocdtrekkie** [comment]
On the upside, since authors are including their public PGP signature in the package, not anything Keybase-specific, there should be no changes to the app packages themselves if we switch the validation method, if people use the same private key with whatever alternative we switch to.

In the interests of backwards compatibility, we probably would need or want to continue to attempt to also validate against Keybase as a fallback for a while, especially since a lot of app authors are no longer active in the Sandstorm community.

**Comment by ocdtrekkie** [comment]
An interesting option is Keyoxide: https://keyoxide.org/

A major highlight is that it has users add links to their proofs as notations on the gpg public key itself, rather than depending on a service to keep track of them. This means the Keyoxide site (open source) is mostly just displaying a profile page based on what's included directly in the public key, which can be hosted on standard PGP keyservers like keys.openpgp.org

We'd still need to verify said proofs ourselves though, either from Sandstorm servers or the Sandstorm App Market (perhaps as an API endpoint that would cache data about proof verifications so that Sandstorm servers themselves need not regularly call out to major services like Facebook and Twitter).

**Comment by troyjfarrell** [comment]
While we're throwing out options: https://keys.pub/

This one appears to keep the bones of Keybase.  I haven't thought about it enough to know whether it has kept enough of the bones to be worth considering.

**Comment by zenhack** [comment]
Someone on IRC mentioned that for GitHub you can fetch PGP keys for a user account, e.g:

https://github.com/ocdtrekkie.gpg

Probably for that to work the package would have to actually mention the author's GitHub handle, but it might be usable as part of a mechanism, though obviously it only covers GitHub

**Comment by ocdtrekkie** [comment]
Some notes:

I spoke extensively with the creator of Keyoxide today. They have recently secured a new grant from NLnet and are working on Keyoxide v2, which will probably be a fair bit closer to what we need than Keyoxide is today? Where they are currently, we would need to provide a public key which already contains the proof information, which would not provide any sort of backwards compatibility for existing packages. But they are looking at a new signature profile feature that would be probably cached/queryable such that we might be able to match existing keys with updated profiles and proofs, like we do with Keybase today.

keys.pub appears to possibly no longer be in development, it looks like their last commits were 14 months ago now.

**Comment by xet7** [comment]
@zdb999 @ocdtrekkie @zenhack @troyjfarrell 

It seems keybase.pub part of Keybase is shutting down at March 1 2023, other parts still running:
- https://keybase.pub
- https://news.ycombinator.com/item?id=34709699
